app/models/base_organisation.rb

```diff
@@ -54,5 +57,32 @@ def setup_categories
   def has_relation?(category)
     self.category_organisations.any? {|v| v.category_id == category.id}
   end
+  
+  def slug_candidates
+    [
+      :short_name,
+      :prolonged_name,
+      :orged,
+      :name
+    ]
+  end
+  
+  def short_name
+    slug_words.first(3).join('-')
+  end
+  
+  def prolonged_name
+    "#{self.short_name}-#{slug_words[-2]}-#{slug_words[-1]}"
+  end
+  
+  def orged
+    "#{slug_words.join('-')}-org"
+  end
+  
+  def slug_words
+    not_wanted = ['the', 'of', 'for', 'and', 'in', 'to']
+    words = "#{self.name}".delete('\'','-').scan(/\b\w+\b/).map(&:downcase)
+    words.reject {|w| not_wanted.include?(w) }
+  end
```

Comment:
these last four methods are all string manipulation methods, could they be pulled out of base org and into a utility class (possibly even String)?  Then they could be tested separately from the base org ...